### PR TITLE
Expose ParseOptions on generator context

### PR DESCRIPTION
**Pull request by chsienki**
Small PR that exposes the parse options that will be used to the consuming generator

### CODE REVIEW

**Comment by chsienki** [comment]
@dotnet/roslyn-compiler for review please

File: src/Compilers/Core/Portable/SourceGeneration/SourceGeneratorContext.cs [hunk 1]
```diff
@@ -41,6 +42,11 @@ internal SourceGeneratorContext(Compilation compilation, ImmutableArray<Addition
         /// </remarks>
         public Compilation Compilation { get; }
 
+        /// <summary>
+        /// Get the <see cref="ParseOptions"/> that will be used to parse any added sources.
+        /// </summary>
+        public ParseOptions ParseOptions { get; }
```
Comment (jcouv) [inline src/Compilers/Core/Portable/SourceGeneration/SourceGeneratorContext.cs:48]: Can you add tests for the added public API?

Also, just for my context, can you clarify where those parse options come from (how are they computed)?

Comment (chsienki) [reply src/Compilers/Core/Portable/SourceGeneration/SourceGeneratorContext.cs:48]: Added a test. The options are passed to the generator driver at construction time. In CSC we pass them here: https://github.com/dotnet/roslyn/blob/5a609283e57b2f1ffc1011a1467afe1edf5ff2cd/src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs#L961 

Which come directly from the command line args.

**Comment by chsienki** [comment]
Ping @dotnet/roslyn-compiler for a second review please :)

**Comment by 333fred** [comment]
I'm not sure why this is necessary: if an analyzer author cares about the parse options, they have access to the compilation. They can either get a ParseOptions off one of the trees, or more likely, answer any questions by calling methods on the Compilation directly.

**Comment by chsienki** [comment]
@333fred suggested we just add the `ParseOptions` to the compilation directly. 

My thought is that generally on a compilation they aren't that interesting, as you can just ask questions of the compilation about 'what was parsed'. With generators though the parse options become more interesting, because it's saying 'how is the compiler going to parse this text I give it'. 

We could move it to the compilation, but given the above, I don't see that it's particularly useful anywhere other than in a generator. (You can also always just go and get the options via the trees if you need to. This PR just makes it simpler/more obvious for generators to do so).

**Comment by CyrusNajmabadi** [comment]
Hrmm... can you clarify how this would work?  ParseOptions are specific to a tree, not the full compilation.  So how can you expose something at teh compilation level?

**Pull request merged**